pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -1299,20 +1302,54 @@ func (p *PriorityQueue) PendingPods() ([]*v1.Pod, string) {
 	return result, fmt.Sprintf(pendingPodsSummary, p.activeQ.Len(), p.podBackoffQ.Len(), len(p.unschedulablePods.podInfoMap))
 }
 
+func (p *PriorityQueue) nominatedPodToInfo(np NominatedPod) *framework.PodInfo {
+	pod := np.ToPod()
+	pInfoLookup := newQueuedPodInfoForLookup(pod)
+
+	obj, exists, _ := p.activeQ.Get(pInfoLookup)
+	if exists {
+		queuedPodInfo := obj.(*framework.QueuedPodInfo)
+		return queuedPodInfo.PodInfo
+	}
+
+	queuedPodInfo := p.unschedulablePods.get(pod)
+	if queuedPodInfo != nil {
+		return queuedPodInfo.PodInfo
+	}
+
+	obj, exists, _ = p.podBackoffQ.Get(pInfoLookup)
+	if exists {
+		queuedPodInfo := obj.(*framework.QueuedPodInfo)
+		return queuedPodInfo.PodInfo
+	}
+
+	return &framework.PodInfo{Pod: pod}
+}
+
+func (p *PriorityQueue) nominatedPodsToInfo(nominatedPods []NominatedPod) []*framework.PodInfo {
+	p.qLock.RLock()
+	defer p.qLock.RUnlock()
+	pods := make([]*framework.PodInfo, len(nominatedPods))
+	for i, np := range nominatedPods {
+		pods[i] = p.nominatedPodToInfo(np).DeepCopy()
+	}
+	return pods
+}
+
 // Close closes the priority queue.
 func (p *PriorityQueue) Close() {
-	p.lock.Lock()
-	defer p.lock.Unlock()
+	p.qLock.Lock()
+	defer p.qLock.Unlock()
 	close(p.stop)
 	p.closed = true
 	p.cond.Broadcast()
 }
 
 // DeleteNominatedPodIfExists deletes <pod> from nominatedPods.
 func (npm *nominator) DeleteNominatedPodIfExists(pod *v1.Pod) {
-	npm.lock.Lock()
+	npm.nominatorLock.Lock()
 	npm.deleteNominatedPodIfExistsUnlocked(pod)
-	npm.lock.Unlock()
+	npm.nominatorLock.Unlock()
 }
 
 func (npm *nominator) deleteNominatedPodIfExistsUnlocked(pod *v1.Pod) {
```

Comment:
I guess we don't need `deleteNominatedPodIfExistsUnlocked` anymore because it's only used from DeleteNominatedPodIfExists.

```go
// DeleteNominatedPodIfExists deletes <pod> from nominatedPods.
func (npm *nominator) DeleteNominatedPodIfExists(pod *v1.Pod) {
	npm.nominatorLock.Lock()
	npm.delete(pod)
	npm.nominatorLock.Unlock()
}
```

Reply:
Did the same with `updateNominatedPodUnlocked`